### PR TITLE
Add a configuration section and improve the usage section

### PR DESCRIPTION
**Pull request by Mogztter**
Review will be highly appreciated!

Ping @eshepelyuk  @danyill

### CODE REVIEW

**Comment by eshepelyuk** [comment]
As a general comment, why not having README in AsciiDoct itself ? Only for sake of displaying README in NPM ?

**Comment by Mogztter** [comment]
> As a general comment, why not having README in AsciiDoct itself ? Only for sake of displaying README in NPM ?

Yes and I'm not pleased... just look at this ugly table with inline HTML :nauseated_face: 

> As a suggestion - I would move Antora Integration into Install section, since I consider there're 3 different way of installation: nodejs, browser and antora.

That's a good idea :+1: 

> As a 2nd suggestion - I'd rename Usage to Supported diagram types

Not sure about this one. We could move the list of supported diagrams in "Supported diagram types" but I want to keep the Usage section to explain the syntax (macro, literal block), the attributes (target, format, generated file name...) and the options (inline, interactive...).

**Comment by eshepelyuk** [comment]
>> As a 2nd suggestion - I'd rename Usage to Supported diagram types

> Not sure about this one. We could move the list of supported diagrams in "Supported diagram > types" but I want to keep the Usage section to explain the syntax (macro, literal block), the attributes (target, format, generated file name...) and the options (inline, interactive...).

Well, I suggest `Configuration` (i.e. supported attributes) should be the part of `Usage` in such case, along with beforementioned settings.

File: README.md [hunk 1]
```diff
@@ -65,24 +66,45 @@ Create a file named `kroki.html` with the following content and open it in your
...
 ## Usage
 
-```adoc
-[plantuml,alice-bob,svg,role=sequence]
-....
-alice -> bob
-....
+Kroki currently supports the following diagram libraries:
+
+* ActDiag: `actdiag`
```
Comment (danyill) [inline README.md:82]: It's probably a bit of effort but as an end user who is not familiar with all of these I would like each one to be a hyperlink to the library location.

File: README.md [hunk 2]
```diff
@@ -65,24 +66,45 @@ Create a file named `kroki.html` with the following content and open it in your
...
+* PlantUML: `plantuml`
+* RackDiag: `rackdiag`
+* SeqDiag: `seqdiag`
+* SVGBob: `svgbob`
+* UMLet: `umlet`
+* Vega: `vega`
+* Vega-Lite: `vegalite`
+* WaveDrom: `wavedrom`
+
+In your AsciiDoc document, you can either declare your diagram inline or reference a diagram file using the corresponding macro or using the `include` directive.
```
Comment (danyill) [inline README.md:103]:
```suggestion
In your AsciiDoc document, you can either write your diagram inline or alternatively you can make a reference to the diagram file using macro form or with the `include` directive.
```
Too many "ors" running together, just a suggestion...

File: README.md [hunk 3]
```diff
@@ -96,6 +118,36 @@ digraph foo {
...
+vegalite::chart.vlite[svg,role=chart,opts=interactive]
+```
+
+![Vega-Lite chart diagram](https://kroki.io/vegalite/png/eNrtVktz2yAQvvtXMEyOqt9pnNz6To-d6c3jA5ZWEg0CF7Ba26P_3gVb2JJSN8mhTWdyMIb92CffCnY9QuiFiXMoGL0hNLd2ZW4GgxIy1s-4zdfLPleD_QYvfSW4hUE57X8zStLI6SdgYs1XlqMAbdwqzbdKWibEhsRKxsyCxF9C4pxpa4jNmSUmVz9IwtMUNEhL7GYFhqgURWgMLN9ymRETMwGmf3DDrItxh3NclUysweB67teE7KjP4A2NCF3ibDyroib0toYuL9vQuxqaTtrQ-xq6HrWhDzU060Afg6-OwU81NLpuQ7fB4FUb-hwMjiuPLHD0m2i-L3Koxe6gSQum75xuzHUsgNYWKchYJVjfUE0v3TSWKEg5iMTpL4Oql7uzcmKpCi6ZaIJGaReJXAvRkLOf3LQcOFM8vnPilAkDURNLVMG4_A1ouRVw8HOCVGFeHRWo4Vt4bHLf10yiE2Z5Ca0MHSnvSaWhiA7_GFashNJ_P65WJbegFeJWr-E04oZpARnI5L7j258C_XI-6d7p_8H0C0v_PUtFhw2aycxtmM-GERm9xmE8xWEyxmE6HC6eJam7afgLy-8oWIZX26OZnSpd-E8qTWh0lvTihfT_C-ltrgHfHaJzpCGf-QR5fjVcnOuK8XDfEM-tF56c3bFZSq45PsDo0y-CryGIhzQFjj4YikpKlMfkOrmGWlIuE1hhEPhqPLbNgUYNMLioelXvF-H7eDo=)
+
+
+Finally, we can use the `include` directive to reference a diagram file:
+
+```
+[plantuml,alice-bob,svg,role=sequence]
```
Comment (danyill) [inline README.md:135]: We need to define which attributes are available and what they do somewhere. Perhaps I can do this as part of #72 before too long.

**Comment by marcelstoer** [comment]
This would fix #39?

File: README.md (hunk 2 above)
Comment (Mogztter) [reply README.md]: Thanks that's better!

File: README.md (hunk 3 above)
Comment (Mogztter) [reply README.md:135]: Yes that would be nice :smile:

**Comment by Mogztter** [comment]
> This would fix #39?

@marcelstoer Partially, as mentioned by @danyill  we will still need to define which attributes are available and what they do (ie. https://github.com/asciidoctor/asciidoctor-diagram#diagram-attributes)

File: README.md [hunk 4]
```diff
@@ -96,6 +95,64 @@ digraph foo {
...
+* [PlantUML](https://github.com/plantuml/plantuml): `plantuml`
+* [RackDiag](https://github.com/blockdiag/nwdiag): `rackdiag`
+* [SeqDiag](https://github.com/blockdiag/seqdiag): `seqdiag`
+* [SVGBob](https://github.com/ivanceras/svgbob): `svgbob`
+* [UMLet](https://github.com/umlet/umlet): `umlet`
+* [Vega](https://github.com/vega/vega): `vega`
+* [Vega-Lite](https://github.com/vega/vega-lite): `vegalite`
+* [WaveDrom](https://github.com/wavedrom/wavedrom): `wavedrom`
+
+Each diagram libraries support one or more output format.
```
Comment (danyill) [inline README.md:176]:
```suggestion
Each diagram libraries support one or more output formats.
```

File: README.md [hunk 5]
```diff
@@ -96,6 +95,64 @@ digraph foo {
...
+Consult the [Kroki documentation](https://kroki.io/#support) to find out which format are supported.
```
Comment (danyill) [inline README.md:177]:
```suggestion
Consult the [Kroki documentation](https://kroki.io/#support) to find out which formats are supported.
```

Comment (Mogztter) [reply README.md]: Good catch! :sweat:

**Comment by Mogztter** [comment]
Thanks for your review, that was really helpful!
I think I've addressed all your suggestions, feel free to review one more time before I merge :wink:

**Comment by danyill** [comment]
Just for me (and then we can close #81) can we change the browser example to begin with:

```html
<!DOCTYPE html>
  <head>
    <meta charset="utf-8">
```

Then we're done (and we are reminded of why we do not love markdown!)

**Comment by danyill** [comment]
Oh I can't resist. Can we change the browser example to something which works like:

```
      var input = `
[plantuml,alice-bob,svg,role=sequence]
....
alice -> bob
....

[graphviz]
....
digraph foo {
  node [style=rounded]
  node1 [shape=box]
  node2 [fillcolor=yellow, style="rounded,filled", shape=diamond]
  node3 [shape=record, label="{ a | b | c }"]
  node1 -> node2 -> node3
}
....
`
```

**Comment by Mogztter** [comment]
Based on @danyill feedback, I overhauled the "Browser" section to document what needs to be done if you want to use the macro form.

**Comment by danyill** [comment]
This is a great README! :100: 

Thanks :clap: :clap:

**Comment by Mogztter** [comment]
Thanks again for your kind review, let's merge!

**Pull request merged**